practice/vue-app/src/components/BaseButton.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <button
+    class="rounded-md text-lg"
+    :class="buttonClass"
+  >
+    <slot></slot>
+  </button>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue';
+
+const { validate, reset } = defineProps({
+  validate: {
+    type: Boolean,
+    required: false
+  },
+  reset: {
+    type: Boolean,
+    required: false
+  }
+});
+
+const validateButtonClass = "bg-blue-200 hover:bg-blue-800 hover:text-white p-4 mt-4"
+const resetButtonClass = "bg-red-200 hover:bg-red-800 hover:text-white p-4 mt-4"
+
+const buttonClass = computed<string>(() => {
+  if (validate) {
+    return validateButtonClass;
+  } else if (reset) {
+    return resetButtonClass;
+  } else {
+    return "";
+  }
+});

```

Comment:
Wprowadzenie BaseButtona to dobra decyzja - zakładam, że miałeś na myśli wprowadzenie reużywalnego komponentu prezentacyjnego.  Fakt, że ten button wie o scenariuszach biznesowych - reset i validate - jest złamaniem oryginalnego założenia (stąd pusta klasa w gałęzi else). 

Zaproponowałbym dwa alternatywne podejścia:
1) BaseButton przyjmuje dowolny zestaw klas i dodaje je do swojego template'u
2) Wprowadzenie komponentów ValidateButton i ResetButton, zdefiniowanie ich klas i reużycie w środku pomysłu nr 1

Reply:
Ajj zdecydowanie, masz racje. Niepotrzebnie pomieszałem scenariusze biznesowe z button, który miał być reużywalnym komponentem.
Tutaj jest pull request na mojego brancha master ze zmianami:
[https://github.com/lewpiotr1701/opanuj-frontend-praktyka/pull/2](url)